src/ProxyManager/ProxyGenerator/LazyLoadingValueHolderGenerator.php

```diff
@@ -52,9 +55,14 @@ class LazyLoadingValueHolderGenerator implements ProxyGeneratorInterface
      *
      * @throws InvalidProxiedClassException
      * @throws InvalidArgumentException
+     *
+     * @psalm-param array{skipDestructor?: bool} $proxyOptions
      */
-    public function generate(ReflectionClass $originalClass, ClassGenerator $classGenerator)
+    public function generate(ReflectionClass $originalClass, ClassGenerator $classGenerator/*, array $proxyOptions = []*/)
     {
+        /** @psalm-var array{skipDestructor?: bool} $proxyOptions */
+        $proxyOptions = func_num_args() >= 3 ? func_get_arg(2) : [];
```

Comment:
IMO BC break is a better approach here, and preferable to this sort of API design.

Reply:
This one is also tripping the mutation test, as far as I can see